### PR TITLE
#1531: Note the unresolved dependencies

### PR DESCRIPTION
**Pull request by troydai**
Issue: #1531 

/cc @davidfowl 
/review: @Praburaj @victorhurdugaci @muratg @BrennanConroy

### CODE REVIEW

**Comment by troydai** [comment]
image

File: src/Microsoft.Framework.PackageManager/List/LibraryDependencyFinder.cs [hunk 1]
```diff
@@ -10,32 +12,34 @@ namespace Microsoft.Framework.PackageManager.List
...
-        }
+            var libDictionary = libraries.ToDictionary(desc => desc.Identity);
 
-        public IGraphNode<Library> Build(Library root)
-        {
-            return DepthFirstGraphTraversal.PostOrderWalk<Library, IGraphNode<Library>>(
+            LibraryDescription root;
+            if (!libDictionary.TryGetValue(new Library { Name = project.Name, Version = project.Version }, out root))
+            {
+                throw new InvalidOperationException(string.Format("Fail to retrieve {0} of project {1} - {2}", typeof(LibraryDependency).Name, project.Name, project.Version));
```
Comment (BrennanConroy) [inline src/Microsoft.Framework.PackageManager/List/LibraryDependencyFinder.cs]: "Failed"

**Comment by troydai** [comment]
:watch:

File: src/Microsoft.Framework.PackageManager/List/LibraryDependencyFinder.cs [hunk 2]
```diff
@@ -10,32 +12,34 @@ namespace Microsoft.Framework.PackageManager.List
 {
     public class LibraryDependencyFinder
     {
-        private readonly IDictionary<Library, LibraryDescription> _libDictionary;
-
-        public LibraryDependencyFinder(IDictionary<Library, LibraryDescription> libDictionary)
+        public static IGraphNode<LibraryDescription> Build([NotNull] IList<LibraryDescription> libraries, 
+                                                           [NotNull]Runtime.Project project)
```
Comment (BrennanConroy) [inline src/Microsoft.Framework.PackageManager/List/LibraryDependencyFinder.cs]: nit: Space after [NotNull]

**Comment by BrennanConroy** [comment]
Looks good to me.

**Comment by wangkanai** [comment]
Does this solve #1642 The system cannot find the path specified

**Pull request merged**